main.js

```diff
@@ -294,6 +321,25 @@ export class IdDecoder {
         bitLength: this.fixedBitLength
       });
     }
+    if(this.multihash) {
+      // <varint hash fn code>: identity function
+      const [hashFnCode] = decoded.slice(0, 1);
+      if(hashFnCode !== MULTIHASH_IDENTITY_FUNCTION_CODE) {
+        throw new Error('Invalid multihash function code.');
+      }
+      // <varint digest size in bytes>: 32
+      const [digestSize] = decoded.slice(1, 2);
```

Comment:
Is this the same?

```suggestion
      const digestSize = decoded[1];
```

Given that we're not actually properly decoding a varint here we should probably make note of that -- or properly decode it.

Reply:
Updated